cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1271,46 +1271,53 @@ public void getStatuses(FullHttpRequest request, HttpResponder responder,
   @Path("/stop")
   @AuditPolicy({AuditDetail.REQUEST_BODY, AuditDetail.RESPONSE_BODY})
   public void stopPrograms(FullHttpRequest request, HttpResponder responder,
-                           @PathParam("namespace-id") String namespaceId) throws Exception {
-
+                           @PathParam("namespace-id") String namespaceId,
+                           @QueryParam("graceful") String gracefulShutdownSecs) throws Exception {
+    LOG.info("Stopping all programs in namespace {}", namespaceId);
```

Comment:
was this added intentionally? Seems unrelated

Reply:
I made changes to this file and the ProgramLifecycleService files to validate my changes. Will remove this commit once I am done testing